src/Evolution/Systems/GeneralizedHarmonic/Characteristics.cpp

```diff
@@ -25,10 +27,15 @@ void characteristic_speeds(
     const gsl::not_null<std::array<DataVector, 4>*> char_speeds,
     const Scalar<DataVector>& gamma_1, const Scalar<DataVector>& lapse,
     const tnsr::I<DataVector, Dim, Frame>& shift,
-    const tnsr::i<DataVector, Dim, Frame>& unit_normal_one_form) {
+    const tnsr::i<DataVector, Dim, Frame>& unit_normal_one_form,
+    const std::optional<tnsr::I<DataVector, Dim, Frame>>& mesh_velocity) {
   const auto shift_dot_normal = get(dot_product(shift, unit_normal_one_form));
   (*char_speeds)[0] =
       -(1. + get(gamma_1)) * shift_dot_normal;  // lambda(VSpacetimeMetric)
+  if (mesh_velocity.has_value()) {
+    (*char_speeds)[0] -=
+        get(gamma_1) * get(dot_product((*mesh_velocity), unit_normal_one_form));
```

Comment:
Good point that I hadn't considered. It's much easier to just leave it the way it was and add the damping mesh velocity correction to upwind penalty separate from the free function, so that's the way I think I should go, unless anyone objects

Reply:
That's good with me!